### PR TITLE
Update pre-commit

### LINKED ISSUE #34: More pre-commit
Taking some inspiration from poetry https://github.com/python-poetry/poetry/blob/master/.pre-commit-config.yaml I suggest to update our pre-commit config
1. by opting into
    ```
      - id: check-merge-conflict
      - id: check-case-conflict
      - id: check-ast
      - id: check-docstring-first
    ```
2. adding
   ```
    - repo: https://github.com/pre-commit/pre-commit
      rev: v2.20.0
      hooks:
        - id: validate_manifest
   ```
3. opting into https://pre-commit.ci/ (with `autofix_prs: false`) to enforce people running pre-commit (right @andreab1997 ? :upside_down_face: )

what do you think? @AleCandido @andreab1997 @niclaurenti @giacomomagni

### PR DESCRIPTION
**Pull request by felixhekhorn**
Closes #34

### CODE REVIEW

**Review by AleCandido — APPROVED** [review]

**Review by AleCandido — CHANGES_REQUESTED** [review]

File: src/pineko/configs.py [hunk 1]
```diff
@@ -6,13 +6,12 @@
 import appdirs
 import tomli
 
+# Name of the config file (wherever it is placed)
 name = "pineko.toml"
-"Name of the config while (wherever it is placed)"
```
[the remark this answers is not in the document]
Comment (AleCandido) [reply src/pineko/configs.py:10]: Sphinx recognize them:
https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html

[the remark this answers is not in the document]
Comment (felixhekhorn) [reply src/pineko/configs.py:10]: actually I read only the first half - seems with `#: bla` sphinx can catch them nevertheless

[the remark this answers is not in the document]
Comment (AleCandido) [reply src/pineko/configs.py:10]: For me `pydocstring` didn't complain, I readded them

[the remark this answers is not in the document]
Comment (felixhekhorn) [reply src/pineko/configs.py:10]: and I'm sorry it is not pydocstyle - but it was pre-commit himself

[the remark this answers is not in the document]
Comment (AleCandido) [reply src/pineko/configs.py:10]: > and I'm sorry it is not pydocstyle - but it was pre-commit himself

Ok, but then `check docstring is first` is conflicting with Sphinx style guide, so I'd stick to a single one, and drop that check

[the remark this answers is not in the document]
Comment (felixhekhorn) [reply src/pineko/configs.py:10]: done in https://github.com/N3PDF/pineko/commit/76adb7cccd68791dbe5b3a3aef1f9093ca288908

**Pull request merged**